app/models/profile.rb

```diff
@@ -3,12 +3,35 @@ class Profile < ApplicationRecord
 
   validates :user_id, uniqueness: true
 
-  # This method generates typed accessors for all profile fields
-  def self.refresh_store_accessors!
+  # NOTE: @citizen428 This is a temporary mapping so we don't break DEV during
+  # profile migration/generalization work.
+  MAPPED_ATTRIBUTES = {
+    brand_color1: :bg_color_hex,
+    brand_color2: :text_color_hex,
+    display_email_on_profile: :email_public,
+    display_looking_for_work_on_profile: :looking_for_work_publicly,
+    git_lab_url: :gitlab_url,
+    linked_in_url: :linkedin_url,
+    recruiters_can_contact_me_about_job_opportunities: :contact_consent,
+    stack_overflow_url: :stackoverflow_url
+  }.with_indifferent_access.freeze
+
+  # Generates typed accessors for all currently defined profile fields.
+  def self.refresh_attributes!
     ProfileField.find_each do |field|
       store_attribute :data, field.attribute_name, field.type
     end
   end
 
-  refresh_store_accessors!
```

Comment:
We do no longer call this automatically. It breaks things in horrible ways when no DB exists yet and we try to create one (e.g. in CI or whenever setting up the app from scratch). Instead, we now call `refresh_attributes!` explicitly in the service object and other relevant places.